dGame/dCinema/Prefab.cpp

```diff
@@ -0,0 +1,172 @@
+#include "Prefab.h"
+
+#include "tinyxml2.h"
+
+#include "../../dWorldServer/ObjectIDManager.h"
+#include "EntityManager.h"
+#include "EntityInfo.h"
+#include "RenderComponent.h"
+
+using namespace Cinema;
+
+std::unordered_map<std::string, Prefab> Prefab::m_Prefabs;
+std::unordered_map<size_t, Prefab::Instance> Prefab::m_Instances;
+
+size_t Prefab::AddObject(LOT lot, NiPoint3 position, NiQuaternion rotation, float scale) {
+	const auto id = ObjectIDManager::GenerateRandomObjectID();
+
+	m_Pieces.emplace(id, Prefab::Piece { lot, position, rotation, scale, {} });
```

Comment:
Personally not a fan of the {} without a struct name because you dont know what it is here out of context, Id recommend adding the class name before that so its more readable